### DISCUSSION TITLE
Geojson types

### DISCUSSION THREAD

**Discussion by kaligrafy** [visgl/deck.gl#9082] [Q&A]
Hi!

When I use typescript and I wish to type a featureCollection, I usually use the default GeoJSON.FeatureCollection<GeoJSON.MultiPolygon> or other feature type, but it fails to match the deck.gl internal geojson types and I get errors like:

Type 'FeatureCollection<MultiPolygon, GeoJsonProperties>' is not assignable to type 'FeatureCollection'.

Do you know how to make deck.gl types compatible with default ones?

**Comment by ibgreen** [comment]
>Do you know how to make deck.gl types compatible with default ones?

What does "default" mean to you? deck.gl is importing a type library for these so they are not home grown.

https://github.com/visgl/deck.gl/blob/6971bc6dcaa81a1ab5963bfdac85c7b13d22c3ef/modules/geo-layers/package.json#L54

In fact there was recently a PR in deck.gl-community to align with the above: https://github.com/visgl/deck.gl-community/pull/110

**Reply by kaligrafy** [reply]
If you use the type: GeoJSON.FeatureCollection<GeoJSON.MultiPolygon> without imoprting any type, it works out of the box. I guess these are now included by default in typescript or am I missing something?

**Reply by ibgreen** [reply]
Interesting. Unfortunately, I can't find any evidence of this, including in the typescript playground. Have you checked your projects `tsconfig.json`? it can reference and "inject" additional types.

**Comment by kaligrafy** [comment]
I see these are from 'geojson'
import GeoJSON from 'geojson';

**Reply by ibgreen** [reply]
There is a discussion about the `geojson` module in the comments in the deck.gl-community repo, I did not see any types, rather it exports code to transform geojson code that we are not particularly interested in.

**Reply by ibgreen** [reply]
I guess `@types/geojson` are intended as types for the `geojson` module.

**Comment by kaligrafy** [comment]
So we just need to wait until the geojson types are used instead of custom ones in https://github.com/visgl/deck.gl-community/pull/110

**Reply by ibgreen** [reply]
Well, only if you are using any modules from deck.gl-community. Those modules are optional, unsupported add-ons to deck.gl.

**Comment by kaligrafy** [comment]
Thanks!
